docs: improve app.config.ts documentation for importing .ts files (#25784)
expo/expo: docs/pages/guides/typescript.mdx

**app.config.ts** is supported by default. However, it doesn't support external TypeScript modules, or **tsconfig.json** customization. You can use the following approach to get a more comprehensive TypeScript setup:

```ts app.config.ts

> import 'ts-node/register' // Add this to import TypeScript files

Reviewer: is this import required? i don't use it here: https://github.com/brentvatne/microfoam-app/blob/main/apps/mobile/app.config.ts

> import 'ts-node/register' // Add if you want to import TypeScript files

Reviewer 2: ```suggestion
import 'ts-node/register' // Add this to import TypeScript files
```
